hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/ha/OMFailoverProxyProvider.java

```diff
@@ -70,26 +71,46 @@
   private final UserGroupInformation ugi;
   private final Text delegationTokenService;
 
+  // TODO: Do we want this to be final?
+  private String omServiceId;
+
   public OMFailoverProxyProvider(OzoneConfiguration configuration,
-      UserGroupInformation ugi) throws IOException {
+      UserGroupInformation ugi, String omServiceId) throws IOException {
     this.conf = configuration;
     this.omVersion = RPC.getProtocolVersion(OzoneManagerProtocolPB.class);
     this.ugi = ugi;
-    loadOMClientConfigs(conf);
+    this.omServiceId = omServiceId;
+    loadOMClientConfigs(conf, this.omServiceId);
     this.delegationTokenService = computeDelegationTokenService();
 
     currentProxyIndex = 0;
     currentProxyOMNodeId = omNodeIDList.get(currentProxyIndex);
   }
 
-  private void loadOMClientConfigs(Configuration config) throws IOException {
+  public OMFailoverProxyProvider(OzoneConfiguration configuration,
+      UserGroupInformation ugi) throws IOException {
+    this(configuration, ugi, null);
+  }
+
+  private void loadOMClientConfigs(Configuration config, String omSvcId)
+      throws IOException {
     this.omProxies = new HashMap<>();
     this.omProxyInfos = new HashMap<>();
     this.omNodeIDList = new ArrayList<>();
 
-    Collection<String> omServiceIds = config.getTrimmedStringCollection(
-        OZONE_OM_SERVICE_IDS_KEY);
+    Collection<String> omServiceIds;
+    if (omSvcId == null) {
```

Comment:
```
if (omSvcId == null) {
      // When no OM service id is passed in
      // Note: this branch will only be followed when omSvcId is null,
      // meaning the host name/service id provided by user doesn't match any
      // ozone.om.service.ids on the client side. Therefore, in this case
      // just treat it as non-HA by assigning an empty list to omServiceIds
      omServiceIds = new ArrayList<>();
    } else {
      omServiceIds = Collections.singletonList(omSvcId);
    }
```

Can be replaced with `omServiceIds = Collections.singletonList(omSvcId)`;

Reply:
Thinking more, we can refactor this below first loop, we don't need at all. You can take up this refactor in a new jira also.

Reply:
@bharatviswa504 You are right. We can remove the condition here completely as it makes no difference.

Will do the refactoring in a new jira.

Reply:
Filed https://issues.apache.org/jira/browse/HDDS-2104